lib/src/stream_extensions.dart

```diff
@@ -78,4 +80,71 @@ extension StreamExtensions<T> on Stream<T> {
       ..onCancel = subscription.cancel;
     return controller.stream;
   }
+
+  /// Invoke [each] for each item in this stream, and wait for the [Future]
+  /// returned by [each] to be resolved. Running no more than [concurrency]
+  /// number of [each] calls at the same time.
+  ///
+  /// This function will wait for the futures returned by [each] to be resolved
+  /// before completing. If any [each] invocation throws, [boundedForEach] stop
+  /// subsequent calls
+  ///
+  /// will
+  /// call [onError], if [onError] throws (it does)
+  /// continue subsequent [each] calls, ignore additional errors and throw the
+  /// first error encountered.
+  Future<void> boundedForEach(
+    int concurrency,
+    FutureOr<void> Function(T item) each, {
+    FutureOr<void> Function(Object e, StackTrace? st) onError = _throwOnError,
```

Comment:
`_throwOnError` -> `_Future<void>.error`. Just use the constructor tear-off. It's constant.